ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/EarlyExit.hs

```diff
@@ -56,6 +57,18 @@ newtype WithEarlyExit m a = WithEarlyExit {
            , MonadPlus
            )
 
+instance NoThunks (StrictTVar m a)
```

Comment:
None of the DerivingVia newtypes work for this because of the `cast`?

Reply:
Indeed

```
src/ouroboros-consensus/Ouroboros/Consensus/Util/EarlyExit.hs:62:1: error: [GHC-25897]
    • Couldn't match type ‘m’ with ‘WithEarlyExit m’
        arising from a use of ‘ghc-prim-0.10.0:GHC.Prim.coerce’
      ‘m’ is a rigid type variable bound by
        the instance declaration
        at src/ouroboros-consensus/Ouroboros/Consensus/Util/EarlyExit.hs:62:1-78
    • In the expression:
        ghc-prim-0.10.0:GHC.Prim.coerce
          @(NoThunks.Class.Context
            -> StrictTVar m a -> IO (Maybe NoThunks.Class.ThunkInfo))
          @(NoThunks.Class.Context
            -> StrictTVar (WithEarlyExit m) a
            -> IO (Maybe NoThunks.Class.ThunkInfo))
          (noThunks @(StrictTVar m a))
      In an equation for ‘noThunks’:
          noThunks
            = ghc-prim-0.10.0:GHC.Prim.coerce
                @(NoThunks.Class.Context
                  -> StrictTVar m a -> IO (Maybe NoThunks.Class.ThunkInfo))
                @(NoThunks.Class.Context
                  -> StrictTVar (WithEarlyExit m) a
                  -> IO (Maybe NoThunks.Class.ThunkInfo))
                (noThunks @(StrictTVar m a))
      When typechecking the code for ‘noThunks’
        in a derived instance for ‘NoThunks
                                     (StrictTVar (WithEarlyExit m) a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for
        ‘NoThunks (StrictTVar (WithEarlyExit m) a)’
    • Relevant bindings include
        noThunks :: NoThunks.Class.Context
                    -> StrictTVar (WithEarlyExit m) a
                    -> IO (Maybe NoThunks.Class.ThunkInfo)
          (bound at src/ouroboros-consensus/Ouroboros/Consensus/Util/EarlyExit.hs:62:1)
   |
62 | deriving via StrictTVar m a instance NoThunks (StrictTVar (WithEarlyExit m) a)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```